docs/react-v9/contributing/rfcs/react-components/convergence/react-18-upgrade.md

```diff
@@ -0,0 +1,225 @@
+type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
+
+// v18
+type ReactNode =
+  | ReactElement
+  | string
+  | number
+  | Iterable<ReactNode>
+  | ReactPortal
+  | boolean
+  | null
+  | undefined
+  | DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES[keyof DO_NOT_USE_OR_YOU_WILL_BE_FIRED_EXPERIMENTAL_REACT_NODES];
+```
+
+#### Related links
+
+- [DefinitelyTyped: React 18 types](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210)
+- [React v9 Slots Proposed Changes](https://github.com/microsoft/fluentui/pull/31548/files)
+
+#### Initial code changes
+
+> [!NOTE]
+> The initial code changes will be made in a separate PR, and this RFC will be updated with the PR link.
+
+### What is not covered by this RFC
+
+#### React 19
+
+Please note that even though our efforts on this RFC take into consideration React 19 as a future goal, the full support will be discussed in a separate RFC.
+
+#### Usage of new React 18 features
+
+There are many new features in React 18 that we could take advantage of, like the new transition API, batching updates, and more. This RFC does not cover the usage of these new features or upgrades to our components, but it is something that we should consider in the future.
+
+## Areas of impact
+
+There are two distinct areas of impact for this RFC:
+
+- Library release include coding changes, documentation updates, and any other changes required to support React 18, including the breaking changes.
+- Migrate our codebase to React 18, adding testing applications to ensure compatibility with the new version but also for backward compatibility.
+
+## Options
+
+Since the breaking changes are unavoidable, the following options are available to make the transition as smooth as possible:
+
+### Option 1: Release a new major version of Fluent UI
+
+This option involves releasing a new **major** version of Fluent UI, updating the type definitions and making necessary changes to the codebase to accommodate the breaking changes. This will follow semantic versioning and will be a major release, which, as off now, means a release of a version number **10**.
+
+The release will include the necessary type changes, documentation updates and any other changes required. This approach comes with super big challenges.
+
+#### Pros
+
+- 👍 Breaking changes following semantic versioning
+- 👍 This would clearly mark the cut on where Fluent have the support, since it wouldn't be among minor releases
+- 👍 Pull the band-aid and start building the trust that major releases will not be a whole new library
+
+#### Cons
+
+- 👎 The biggest challenge is how we currently communicate our library to partners. We have the explicit (and sometimes implicit) nomenclature "Fluent UI v9" everywhere, including chats, meetings, documents, e-mails, pipelines, folder structures and more. Even the root folder where this RFC document is located receive the react-v9 prefix.\
```

Comment:
the problem is not only with communication - applications often integrate code from various code repositories and this makes the promise of no breaking changes important - Package A comes from repo 1 and uses a new version of the code. Package B comes from repo 2 and uses an old version of the code. eventually both packages come to one app and need to work together, in some cases without the possibility of having multiple versions of dependencies bundled. Coordinating versions across different repos is a difficult challenge.